### PR TITLE
Add a wrapper of list to obtain abstraction

### PR DESCRIPTION
**Pull request by phamvannhatvu**

### CODE REVIEW

File: ListWrapper.h [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#ifndef LIST_WRAPPER_H
```
Comment (HuyDNA) [inline ListWrapper.h:1]: Change header to format _<FILENAME>_MIPS_SIMULATOR_

File: ListWrapper.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    //don't know how the fuck this is not called while another move constructor called
+    ListWrapper<T>(ListWrapper<T> &&wrapper) noexcept : list(wrapper.list) 
+    {
+        
+    }
+
+    auto cbegin() const
+    {
+        return begin();
```
Comment (HuyDNA) [inline ListWrapper.h:72]: - Return list.cbegin() instead of begin()

File: ListWrapper.h [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    auto cend() const
+    {
+        return end();
```
Comment (HuyDNA) [inline ListWrapper.h:77]: Return list.cend() instead of end()

File: ListWrapper.h [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    {
+        
+    }
+
+    ListWrapper<T>(const ListWrapper<T> &wrapper) : list(std::deque<T>(wrapper.list))
+    {
+        std::cout << "Copy constructor";
+    }
+
+    //don't know how the fuck this is not called while another move constructor called
```
Comment (HuyDNA) [inline ListWrapper.h:64]: Giberrish comment

File: ListWrapper.h [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    {
+        return list.begin();
+    }
+
+    auto end()
+    {
+        return list.end();
+    }
+
+    T& operator[] (int id)
```
Comment (HuyDNA) [inline ListWrapper.h:90]: Implement an overloaded constant version of operator[]

File: ListWrapper.h [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    {
+        list.pop_back();
+    }
+
+    void popFront()
+    {
+        list.pop_front();
+    }
+
+    int getSize()
```
Comment (HuyDNA) [inline ListWrapper.h:44]: Add constant to this method

File: ListWrapper.h [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+#include <iterator>
+#include <iostream>
+
+template<class T>
+Class ListWrapper
+{
+private:
+    std::deque<T> list;
+public:
+    void pushBack(T item)
```
Comment (HuyDNA) [inline ListWrapper.h:14]: Parameter type should be const T&

File: ListWrapper.h [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    {
+        list.push_back(item);
+    }
+
+    void pushFront(T item)
```
Comment (HuyDNA) [inline ListWrapper.h:19]: Parameter type should be const T&

File: ListWrapper.h [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+    {
+        list.push_front(item);
+    }
+
+    T getBack()
```
Comment (HuyDNA) [inline ListWrapper.h:24]: No need to implement this method

File: ListWrapper.h [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+    {
+        return list.back(); 
+    }
+
+    T getFront()
```
Comment (HuyDNA) [inline ListWrapper.h:29]: No need to implement this method

File: ListWrapper.h [hunk 11]
```diff
@@ -0,0 +1,97 @@
+#ifndef LIST_WRAPPER_H
+#define LIST_WRAPPER_H
+
+#include <queue>
+#include <iterator>
+#include <iostream>
```
Comment (HuyDNA) [inline ListWrapper.h:6]: Remove this redundant dependency

File: ListWrapper.h [hunk 12]
```diff
@@ -0,0 +1,97 @@
+#ifndef LIST_WRAPPER_H
+#define LIST_WRAPPER_H
+
+#include <queue>
+#include <iterator>
```
Comment (HuyDNA) [inline ListWrapper.h:5]: Remove this redundant dependency

**Review by HuyDNA — CHANGES_REQUESTED** [review]
[ ] Function and class definitions should be in this format:
```c++
<Class/Function> {
.......
}
```
[ ] Eliminate auto return type
[ ] Rename branch name to snake_case format
[ ] Consider
```c++
using std::deque;
```

**Comment by phamvannhatvu** [comment]
> [ ] Eliminate auto return type 

Use auto for flexible foreach iteration

**Pull request closed without merging**